internal/service/rds/cluster_activity_stream_test.go

```diff
@@ -0,0 +1,227 @@
+			return fmt.Errorf("incorrect activity stream status: expected: %s, got: %s", rds.ActivityStreamStatusStarted, aws.StringValue(v.ActivityStreamStatus))
+		}
+
+		if aws.StringValue(v.ActivityStreamMode) != rds.ActivityStreamModeSync && aws.StringValue(v.ActivityStreamMode) != rds.ActivityStreamModeAsync {
+			return fmt.Errorf("incorrect activity stream mode: expected: sync or async, got: %s", aws.StringValue(v.ActivityStreamMode))
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckAWSClusterActivityStreamDestroy(s *terraform.State) error {
+	return testAccCheckAWSClusterActivityStreamDestroyWithProvider(s, acctest.Provider)
+}
+
+func testAccCheckAWSClusterActivityStreamDestroyWithProvider(s *terraform.State, provider *schema.Provider) error {
+	conn := provider.Meta().(*conns.AWSClient).RDSConn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_rds_cluster_activity_stream" {
+			continue
+		}
+
+		var err error
+
+		resp, err := tfrds.FindDBClusterByClusterArn(conn, rs.Primary.ID)
+		if err != nil {
+			// Return nil if the cluster is already destroyed
+			if tfresource.NotFound(err) {
+				return nil
+			}
+			return err
+		}
+
+		if *resp.ActivityStreamStatus != rds.ActivityStreamStatusStopped {
+			return fmt.Errorf("DB Cluster %s Activity Stream still exists with status %s when status %s is expected",
+				rs.Primary.ID, *resp.ActivityStreamStatus, rds.ActivityStreamStatusStopped)
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+func testAccAWSClusterActivityStreamConfigBase(clusterName, instanceName string) string {
+	return fmt.Sprintf(`
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+resource "aws_kms_key" "test" {
+  description             = "Testing for AWS RDS Cluster Activity Stream"
+  deletion_window_in_days = 7
+}
+
+resource "aws_rds_cluster" "test" {
+  cluster_identifier  = "%[1]s"
+  availability_zones  = ["${data.aws_availability_zones.available.names[0]}", "${data.aws_availability_zones.available.names[1]}", "${data.aws_availability_zones.available.names[2]}"]
+  master_username     = "foo"
+  master_password     = "mustbeeightcharaters"
+  skip_final_snapshot = true
+  deletion_protection = false
+  engine              = "aurora-postgresql"
+  engine_version      = "11.9"
+}
+
+resource "aws_rds_cluster_instance" "test" {
+  identifier         = "%[2]s"
+  cluster_identifier = aws_rds_cluster.test.id
+  engine             = aws_rds_cluster.test.engine
+  instance_class     = "db.r6g.large"
+}
+	`, clusterName, instanceName)
+}
+
+func testAccAWSClusterActivityStreamConfig(clusterName, instanceName string) string {
+	return acctest.ConfigCompose(
+		testAccAWSClusterActivityStreamConfigBase(clusterName, instanceName),
+		`
+resource "aws_rds_cluster_activity_stream" "test" {
+  resource_arn = aws_rds_cluster.test.arn
+  kms_key_id   = aws_kms_key.test.key_id
+  mode         = "async"
+
+  depends_on = [aws_rds_cluster_instance.test]
```

Comment:
should avoid this and add some retry logic in create if it needs to wait for the instance to be created.

Reply:
If there's no `depends_on` it can also cause a race condition when deleting the entire stack. i.e. If AWS is busy destroying either `aws_rds_cluster` or `aws_rds_cluster_activity_stream` then the destroy operation of the other will fail stating the cluster is in invalid state to perform the operation.

We can wait at initialization, but to wait at destroy would require changes to the `aws_rds_cluster` resources too.